api/src/main/java/org/openmrs/module/insuranceclaims/api/strategies/consumed/ConsumedItemStrategy.java

```diff
@@ -0,0 +1,94 @@
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.InsuranceClaimConstants.CONSUMABLES_LIST_ITEMS_CONCEPT_UUID;
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.InsuranceClaimConstants.CONSUMED_ITEMS_CONCEPT_UUID;
+import static org.openmrs.module.insuranceclaims.api.service.fhir.util.InsuranceClaimConstants.QUANTITY_CONSUMED_CONCEPT_UUID;
+
+@Component("DefaultConsumedItemStrategy")
+public class ConsumedItemStrategy implements GenericConsumedItemStrategy {
+
+    private ProvidedItemService providedItemService;
+
+    @Override
+    public void addProvidedItems(Obs newObs) throws ConsumedItemException {
+        if (newObs.getConcept().getUuid().equals(CONSUMED_ITEMS_CONCEPT_UUID)) {
+            Set<Obs> obsMembers = newObs.getGroupMembers();
+            if (obsMembers != null) {
+                int unitsConsumed = getNumberOfConsumptions(obsMembers);
+                Obs consumedItem = getConsumedItemConcept(obsMembers);
+                createProvidedItems(consumedItem, unitsConsumed);
+            }
+        }
+    }
+
+    private int getNumberOfConsumptions(Set<Obs> obsMembers) {
+        return obsMembers.stream()
+                .filter(obs -> obs.getConcept().getUuid().equals(QUANTITY_CONSUMED_CONCEPT_UUID))
+                .findFirst()
+                .map(obs -> obs.getValueNumeric())
+                .orElse(1.0)
+                .intValue();
+    }
+
+    private Obs getConsumedItemConcept(Set<Obs> obsMembers) throws ConsumedItemException {
+        return obsMembers.stream()
+                .filter(obs -> obs.getConcept().getUuid().equals(CONSUMABLES_LIST_ITEMS_CONCEPT_UUID))
+                .findFirst()
+                .orElseThrow(() -> new ConsumedItemException("Could not create provided item, no concept found"));
+    }
+
+    private void createProvidedItems(Obs consumedItem, int numberOfConsumptions) throws ConsumedItemException {
+        BigDecimal itemPrice = getConceptPrice(consumedItem.getValueCoded());
+        for (int singleConsumption = 0; singleConsumption < numberOfConsumptions; singleConsumption++ ) {
+            createProvidedItemForObservation(consumedItem, itemPrice);
+        }
+    }
+
+    private void createProvidedItemForObservation(Obs consumedItem, BigDecimal price) {
+        ProvidedItem newProvidedItem = new ProvidedItem();
+
+        Concept consumedItemConcept = consumedItem.getValueCoded();
+        newProvidedItem.setItem(consumedItemConcept);
+
+        int patientId = consumedItem.getPerson().getPersonId();
+        Patient patient = Context.getPatientService().getPatient(patientId);
+        newProvidedItem.setPatient(patient);
+
+        newProvidedItem.setDateOfServed(consumedItem.getObsDatetime());
+        newProvidedItem.setStatus(ProcessStatus.ENTERED);
+        newProvidedItem.setPrice(price);
+
+        providedItemService.saveOrUpdate(newProvidedItem);
```

Comment:
do not save without transaction, as far as I remember I already given you similar feedback previously. Please try to figure it out how to omit those mistake in the future

Reply:
Fixed, added transactional annotation to class